### ISSUE TITLE
Finding data later?

### ISSUE THREAD

**Issue by djb-rh** [farnsworth2008/moar-strava-app#1]
I left it running in a tab after connecting to Strava, but then forgot about it and restarted the browser.  Is that data gone and I have to do it again, or is there some other way to find it?

**Comment by farnsworth2008** [comment]
You task continues to run in the background.   

It keeps running even if you close the browser.

When you next connect to the site via o-auth you will see that progress.

**Comment by djb-rh** [comment]
Yeah, I posted on the TMTB forum thread that I figured that out. There’s an image on the thread and it doesn’t look like it knows about the trails I ride? Like CNF? Or it’s not quite working right.

Also, is there a data cue that would let you split trainer data out from road?


—Donnie

On Apr 6, 2019, at 12:26 PM, Mark Farnsworth <notifications@github.com<mailto:notifications@github.com>> wrote:


You task continues to run in the background.

It keeps running even if you close the browser.

When you next connect to the site via o-auth you will see that progress.

**Comment by farnsworth2008** [comment]
I think I can eventually pick out trainer zift stuff.

Trail areas are currently defined with KML files but down the road I plan to make this user oriented.

The big issue right now is scaling this in a way where it runs fast enough to be useful but cheap enough to avoid becoming a pain.  Right now the big cost is that it's a huge number of geo location calls and database points to track support sub totals for detecting (Road vs. Other) and state/country.  

I am thinking about switching to a model that only scores based on the defined trail areas but it's going to be hit or miss depending on how much I am inspired to work on the effort.   

For now it's actually just a comedy to see how slow it runs on one database server with the cheapest possible configuration from AWS.   I find that I can keep it down to just a few cents a day for running costs but only if it runs slow.   Earlier this week I tried a bigger database but it started costing about $3 per day and that's currently too rich for my budget.

So the site will likely be sorta on semi old sloth mode for a couple of weeks until I come up with an approach.

**Comment by djb-rh** [comment]
If you have time, I’d love a basic primer on how to take the github code and actually set this up from scratch in AWS. From setting up the domain name on. I get the gist of what’s happening, but never done it.

I’m not actually going to DO it, but I’m very curious about possible somewhat similar things. Or just a pointer to a primer, really. I’m in the “few cents a day” category, myself.


—Donnie

On Apr 6, 2019, at 1:07 PM, Mark Farnsworth <notifications@github.com<mailto:notifications@github.com>> wrote:


I think I can eventually pick out trainer zift stuff.

Trail areas are currently defined with KML files but down the road I plan to make this user oriented.

The big issue right now is scaling this in a way where it runs fast enough to be useful but cheap enough to avoid becoming a pain. Right now the big cost is that it's a huge number of geo location calls and database points to track support sub totals for detecting (Road vs. Other) and state/country.

I am thinking about switching to a model that only scores based on the defined trail areas but it's going to be hit or miss depending on how much I am inspired to work on the effort.

For now it's actually just a comedy to see how slow it runs on one database server with the cheapest possible configuration from AWS. I find that I can keep it down to just a few cents a day for running costs but only if it runs slow. Earlier this week I tried a bigger database but it started costing about $3 per day and that's currently too rich for my budget.

So the site will likely be sorta on semi old sloth mode for a couple of weeks until I come up with an approach.

**Comment by farnsworth2008** [comment]
It's an open source project so feel free to git clone it.

I will write up a bit in the README soon because really it's mostly a test bed for API goofing around on Strava and Geo Coding plus playing with some of my library code where I am doing database stuff and rest using some techniques that I am exploring.

I just now glanced at the README and see that it is currently a bit broken as well.

Basically what you would need for any Strava project starts with an account on their dev program:
https://developers.strava.com/

The AWS stuff is totally more complex then I think it should be but they have the lead for cloud stuff.
https://aws.amazon.com/premiumsupport/knowledge-center/create-and-activate-aws-account/

Find me via Facebook or LinkedIn for direct messaging.  If you a Java guy or even just trying to learn Java I don't mind spending some time to help you or others get setup if they are interested in helping out with the project.

**Comment by djb-rh** [comment]
Thanks!  As for FB and LinkedIn, I’m pretty opposed to both, so that’s a no-go.  I’m also not a Java programmer, so it probably doesn’t matter much.  I’m a C guy who can tweak other stuff some, but that’s it.

Thanks for the pointers, though.


—Donnie

On Apr 6, 2019, at 1:31 PM, Mark Farnsworth <notifications@github.com<mailto:notifications@github.com>> wrote:


It's an open source project so feel free to git clone it.

I will write up a bit in the README soon because really it's mostly a test bed for API goofing around on Strava and Geo Coding plus playing with some of my library code where I am doing database stuff and rest using some techniques that I am exploring.

I just now glanced at the README and see that it is currently a bit broken as well.

Basically what you would need for any Strava project starts with an account on their dev program:
https://developers.strava.com/

The AWS stuff is totally more complex then I think it should be but they have the lead for cloud stuff.
https://aws.amazon.com/premiumsupport/knowledge-center/create-and-activate-aws-account/

Find me via Facebook or LinkedIn for direct messaging. If you a Java guy or even just trying to learn Java I don't mind spending some time to help you or others get setup if they are interested in helping out with the project.
